PageToDisplaySearHistoryController.cls

```diff
@@ -0,0 +1,47 @@
+/**
+* @description: Controller for visual force page which is used to display
+                the history or the links of the queries asked by the users.
+* @author: Noopur Naik
+*
+* @revision:
+* Version      Date            Description
+* ------------------------------------------------
+* 1.0          2-July-2017     Initial revision.
+**/
+
+public with sharing class PageToDisplaySearHistoryController {
+    
+    public String queryForFetchingEmailContents = 'SELECT id, Email_Subject__c,Email_Body__c from Email_Handler__c';
```

Comment:
Query Should Always contain limit clause or proper filtering condition which will limit your data.